standards/chronologic-piper.md

```diff
@@ -0,0 +1,124 @@
+# Specification
+
+## Summary
+A standardized interface for interacting with subscriptions.
+
+##  Abstract
+This standard defines a contract interface for interacting with subscription data. The scope of this proposal is end-user / wallet interactions. Creation of the subscriptions is out-of-scope of this ERC.
+
+## Our Thoughts
+From the subscriber's side:
+
+* I want strong guarantees on when I can cancel my subscription.
+* I will normally want payments to happen automatically without any action on my part.
+* In some cases it might be valuable to require an approval process.
+* For dynamically priced subscriptions I want to be able to set limits (require authorization if subscription is more than X).
+
+
+From the providers's side:
+
+* I need the ability to charge a fixed fee per subscription time unit (netflix, pandora, etc).
+* I need the ability to charge a dynamic fee per subscription time unit (aws, twilio, etc).
+* I need to be able to create reasonably accurate forecasts for upcoming subscriptions: Programatic checks that subscription accounts have available balance and that subscription is active.
+
+## Motivation
+Standard that covers all usecases of subscription based model with minimal interface suitable for wallets.
+
+### Interface
+
+#### cancel
+
+Immediately cancels the subscription. Implementations should ensure that any unpaid subscription payments are paid to the provider as part of cancellation to ensure providers are able to let subscriptions fees fill up for arbitrary lengths of time, allowing them to reduce overhead from transaction costs.
+
+```SOLIDITY
+
+/** @dev Cancels the subscription
+  * @return true if cancellation succedded 
+*/
+
+function cancel()
+    public
+    returns (bool success);
+```
+
+#### isFunded
+
+Checkes if there is enough funds available for the provider to collect
+ 
+```SOLIDITY
+
+/** @dev Cancels the subscription
+  * @return true if there is enough funds allowed to the contract to pay for subscription 
+*/
+
+function isFunded()
```

Comment:
As we've discussed on Friday - this might be helpful wrapper but needs further discussion if it's really needed. Alternative is:
1. Wallet gets `payment asset` type (e.g. ERC20).
2. Recognizes the contract type and calls `erc20contract.balanceOf(owner);` or/and `erc20contract.allowance(owner, spender);`

Thoughts?

Reply:
Even in the case of ETH, you probably want to store it in ERC20 compatible "vault" which makes this work by default.

Reply:
This will remove the burden of implementing this functionality outside the smart contract and provides the precise way to see the status of the subscription. 
Underlying implementation will be as you suggested.